### PR TITLE
code review request

### PR DESCRIPTION
**Pull request by atenaiis**
Team:

In this project, the aim is to build an application named Private Events and that behaves similar to Eventbrite. The main goal is to put in practice the main concepts of Associations in rails.

Please follow Readme file instructions.

Cheers.

Atenais Campos

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to events_path, notice: 'Event was successfully created.' }
+        # format.json { render :show, status: :created, location: @user }
+      else
+        format.html { render :new }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```
Comment (ElijahTobs) [inline app/controllers/events_controller.rb]: Your create action is expected to use the #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class EventsController < ApplicationController
+  before_action :logged_in?, except: %i[show index]
+
+  def index
+    @events = Event.all
+  end
```
Comment (ElijahTobs) [inline app/controllers/events_controller.rb]: Your index action is expected to:
* [x] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @user = User.new(user_params)
+    @user.save
+    session[:user_id] = @user.id
+    redirect_to '/welcome'
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @events = Event.all
+  end
```
Comment (ElijahTobs) [inline app/controllers/users_controller.rb]: The show action should:
* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @atenaiis  :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] In the event's show page, the `Attend button` below the Author is always present even when the current user is attending that event. As such, each time I click on that attend button, the user's name appears as a duplicate in the attendance list for that event. Kindly fix that.
image
* [x] The user's show page is supposed to display events the user has created but the column for `Created Events` is displaying **all events**(created by the user and other users). Kindly see to that.

#### Associations

##### User as event creator

  * Migrations
    * [x] events table has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] See inline comments below

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Comment by atenaiis** [comment]
Thanks for that helful review.

I'd work on requested changes, please be so kind as to take another look on my project.

cheers!!

Atenais Campos

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: 

## Hello team, thank you for making the required changes, but unfortunately you have failed to fix all of them, for more information please check below :arrow_down: 

### To highlight:
- **The layout, colors are looking amazing, I know that you have put a lot of work there and great job :+1:**

### To improve:
- [x] `The user's show page is supposed to display events the user has created but the column for Created Events is displaying all events(created by the user and other users). Kindly see to that.` This is still an issue. I created two users and each of them created 2 events and in user/show page under the `Created Events` were showing the events created by the other user as well, kindly fix this.


_Please implement all the above mentioned requirements and ask for another code review whenver you are ready_

### **If you have any question related to the requirements please add a comment here and ping me on slack @GzimAsani**

HAPPY CODING :computer: :desktop_computer:

**Comment by atenaiis** [comment]
Thanks for that quick review.

I've implemented the change requested.

cheers!

**Review by GzimAsani — APPROVED** [review]
# STATUS: ACCEPTED :green_book: :muscle: 

## Hello again, thank you so much for making the required change, your project now looks complete, you may merge it and good luck in the next projects :+1: :1st_place_medal: 

_**PS: Don't forget to submit the project completion form**_

HAPPY CODING :computer: :desktop_computer:

**Pull request merged**